core/src/main/kotlin/riven/core/service/workflow/WorkflowExecutionService.kt

```
@@ -184,53 +178,118 @@ class WorkflowExecutionService(
+        val executionRecords: List<ExecutionSummaryProjection> =
+            workflowExecutionRepository.findExecutionWithNodesByExecutionId(executionId)
+
+        if (executionRecords.isEmpty()) {
+            throw NotFoundException("Workflow execution not found: $executionId")
+        }
+
+        if (executionRecords.any {
+                (it.execution.workspaceId != workspaceId) ||
+                        (it.executionNode.workspaceId != workspaceId) ||
+                        (it.node != null && it.node.workspaceId != workspaceId)
+            }) {
```

detekt (1.23.8)

[warning] This condition is too complex (4). Defined complexity threshold for conditions is set to '4'

(detekt.complexity.ComplexCondition)